s3/s3.go

```diff
@@ -82,54 +84,46 @@ func GetAssumeRoleCredentials(opts S3ClientOpts) (*credentials.Credentials, erro
 	return credentials.NewStaticV4(value.AccessKeyID, value.SecretAccessKey, value.SessionToken), nil
 }
 
+func GetCredentials(opts S3ClientOpts) (*credentials.Credentials, error) {
+	if opts.AccessKey != "" && opts.SecretKey != "" {
+		log.Infof("Creating minio client %s using static credentials", opts.Endpoint)
+		return credentials.NewStaticV4(opts.AccessKey, opts.SecretKey, ""), nil
+	} else if opts.RoleARN != "" {
+		log.Infof("Creating minio client %s using assumed-role credentials", opts.RoleARN)
+		return GetAssumeRoleCredentials(opts)
+	} else if opts.UseSDKCreds {
+		log.Infof("Creating minio client using AWS SDK credentials")
+		return GetAWSCredentials(opts)
+	} else {
+		log.Infof("Creating minio client using IAM role")
+		return credentials.NewIAM(nullIAMEndpoint), nil
+	}
+}
+
 // NewS3Client instantiates a new S3 client object backed
-func NewS3Client(opts S3ClientOpts) (S3Client, error) {
+func NewS3Client(opts S3ClientOpts, ctx context.Context) (S3Client, error) {
```

Comment:
By convention and golang coding guidelines, context objects are supposed to be the first argument to functions.